lib/directory-tree.js

```diff
@@ -1,38 +1,77 @@
 var fs = require('fs');
 var path = require('path');
 
-function directoryTree(basepath, extensions) {
-    var _directoryTree = function (name, extensions) {
-        var stats = fs.statSync(name);
-        var item = {
-            path: path.relative(basepath, name),
-            name: path.basename(name)
-        };
-
-        if (stats.isFile()) {
-            if (extensions &&
-                extensions.length > 0 &&
-                extensions.indexOf(path.extname(name).toLowerCase()) == -1) {
-                return null;
+/**
+ * Looks if any if array values is found in the given string.
+ *     ex.
+ *     array  = ['asdf', 1, 'test.js']
+ *     string = 'this/is/the/test/string.js' // expected return: false
+ *     string = 'this/is/a/test.js' // expected return: true
+*
+ * @param  {string} string  The haystack which you want to search through
+ * @param  {array} array    The needles which you are looking for
+ * @return {boolean}        True if a needle is found, otherwise false.
+ */
+searchSubstring = function(string, array){
+    var found = false;
+    array.forEach( function(value, key){
+        if (string.indexOf( value ) > 0 ) {
+            found = true;
+            return true;
```

Comment:
I think this return statement is meaningless. This loop should definitely have a way to early return.